src/compiler/transformers/test/transpile.ts

```diff
@@ -15,53 +17,67 @@ import { getScriptTarget } from '../transform-utils';
  * @param compilerCtx a compiler context to use in the transpilation process
  * @param beforeTransformers TypeScript transformers that should be applied before the code is emitted
  * @param afterTransformers TypeScript transformers that should be applied after the code is emitted
+ * @param afterDeclarations TypeScript transformers that should be applied
+ * after declarations are generated
+ * @param tsConfig optional typescript compiler options to use
  * @returns the result of the transpilation step
  */
 export function transpileModule(
   input: string,
   config?: d.Config | null,
   compilerCtx?: d.CompilerCtx | null,
   beforeTransformers: ts.TransformerFactory<ts.SourceFile>[] = [],
-  afterTransformers: ts.TransformerFactory<ts.SourceFile>[] = []
+  afterTransformers: ts.TransformerFactory<ts.SourceFile>[] = [],
+  afterDeclarations: ts.TransformerFactory<ts.SourceFile | ts.Bundle>[] = [],
+  tsConfig: ts.CompilerOptions = {}
 ) {
-  const options = ts.getDefaultCompilerOptions();
-  options.isolatedModules = true;
-  options.suppressOutputPathCheck = true;
-  options.allowNonTsExtensions = true;
-  options.removeComments = false;
-  options.noLib = true;
-  options.lib = undefined;
-  options.types = undefined;
-  options.noEmit = undefined;
-  options.noEmitOnError = undefined;
-  options.noEmitHelpers = true;
-  options.paths = undefined;
-  options.rootDirs = undefined;
-  options.declaration = undefined;
-  options.composite = undefined;
-  options.declarationDir = undefined;
-  options.out = undefined;
-  options.outFile = undefined;
-  options.noResolve = true;
-
-  options.module = ts.ModuleKind.ESNext;
-  options.target = getScriptTarget();
-  options.experimentalDecorators = true;
-
-  options.jsx = ts.JsxEmit.React;
-  options.jsxFactory = 'h';
-  options.jsxFragmentFactory = 'Fragment';
-
-  const inputFileName = 'module.tsx';
+  const options: ts.CompilerOptions = {
+    ...ts.getDefaultCompilerOptions(),
+    allowNonTsExtensions: true,
+    composite: undefined,
+    declaration: undefined,
+    declarationDir: undefined,
+    experimentalDecorators: true,
+    isolatedModules: true,
+    jsx: ts.JsxEmit.React,
+    jsxFactory: 'h',
+    jsxFragmentFactory: 'Fragment',
+    lib: undefined,
+    module: ts.ModuleKind.ESNext,
+    noEmit: undefined,
+    noEmitHelpers: true,
+    noEmitOnError: undefined,
+    noLib: true,
+    noResolve: true,
+    out: undefined,
+    outFile: undefined,
+    paths: undefined,
+    removeComments: false,
+    rootDirs: undefined,
+    suppressOutputPathCheck: true,
+    target: getScriptTarget(),
+    types: undefined,
+    // add in possible default config overrides
+    ...tsConfig,
+  };
```

Comment:
this is just the same as what was here before just alphabetized, reformatted as an object literal, and adding in the `tsConfig` param.